pages/clubs-list.jsx

```diff
@@ -0,0 +1,237 @@
+var _ = require('underscore');
+var React = require('react');
+var ModalManagerMixin = require('../mixins/modal-manager');
+var TeachAPIClientMixin = require('../mixins/teach-api-client');
+var HeroUnit = require('../components/hero-unit.jsx');
+var Map = require('../components/map.jsx');
+var ModalAddOrChangeYourClub = require('../components/modal-clubs.jsx');
+var ModalRemoveYourClub = require('../components/modal-clubs-remove.jsx');
+
+var ClubListItem = React.createClass({
+  propTypes: {
+    club: React.PropTypes.object.isRequired,
+    username: React.PropTypes.string,
+    onDelete: React.PropTypes.func,
+    onEdit: React.PropTypes.func,
+    onZoomToLocation: React.PropTypes.func.isRequired
+  },
+  handleLocationClick: function() {
+    this.props.onZoomToLocation(this.props.club);
+  },
+  render: function() {
+    var club = this.props.club;
+    var clubName = club.website ? <a href={club.website}>{club.name}</a> : club.name;
+    var isOwned = (club.owner === this.props.username);
+    var ownerControls = null;
+
+    if (isOwned) {
+      ownerControls = (
+        <p>
+          <button className="btn btn-default btn-xs" onClick={this.props.onEdit.bind(null, club.url)}>
+            <span className="glyphicon glyphicon-pencil"></span> Edit
+          </button>
+          &nbsp;
+          <button className="btn btn-default btn-xs" onClick={this.props.onDelete.bind(null, club.url, club.name)}>
+            <span className="glyphicon glyphicon-trash"></span> Remove
+          </button>
+        </p>
+      );
+    }
+
+    return (
+      <li>
+        <h4>{clubName} <Map.ClubStatusLabel showApproved={isOwned} status={club.status}/></h4>
+        <p><span className="club-location" onClick={this.handleLocationClick}><span className="glyphicon glyphicon-map-marker"/> {club.location.split(',')[0]}</span></p>
+        <p>{club.description}</p>
+        <p><small>Led by <a href={"https://webmaker.org/en-US/search?type=user&q=" + club.owner}>{club.owner}</a></small></p>
+        {ownerControls}
+      </li>
+    );
+  }
+});
+
+var ClubList = React.createClass({
+  COLUMNS: 2,
+  GRID_COLUMNS_PER_ROW: 12,
+  propTypes: {
+    clubs: React.PropTypes.array.isRequired,
+    username: React.PropTypes.string,
+    onDelete: React.PropTypes.func,
+    onEdit: React.PropTypes.func,
+    onZoomToLocation: React.PropTypes.func.isRequired
+  },
+  statics: {
+    Item: ClubListItem
+  },
+  renderColumn: function(key, clubs) {
+    var colClass = 'col-xs-' + (this.GRID_COLUMNS_PER_ROW / this.COLUMNS);
+    return (
+      <div className={colClass} key={key}>
+        <ul className="clubs-list list-unstyled">
+          {clubs.map(function(club, i) {
+            return  <ClubListItem key={i} club={club}
```

Comment:
this is not a good key. keys should identify elements, so should be independent of the ordering the array happens to be in. `key={club.name}` would be a good key, since it's tied to the element we're making, not the ordering of the element's base data in some array.